### ISSUE TITLE
Blank password make backup unterminated

### ISSUE THREAD

**Issue by antoniopaolini** [JackGruber/joplin-plugin-backup#21]
Hi, 
Today, back to the office after vacation, I upgraded Joplin (from 2.1.9 to 2.3.5), the plugins and all the needs.
I try to make manual backup, from the Tools menu, but I don't see the backup file in the destination directory.

Then I started fumbling through directories and I see the temporary file jex and a 0-byte 7zip file.
![image](https://user-images.githubusercontent.com/60228211/130429238-a85c6737-bc25-4a93-85e2-838b4543df09.png)
It stuck in this situation.

Then I looked in the plugin configuration and I see this:
![image](https://user-images.githubusercontent.com/60228211/130429329-12f894ce-5243-438f-b687-cc8ac87e0142.png)

In the log file there's no information (see log pasted at bottom).

I think there's a problem when the password is left blank.
N.B. The password was set before the last joplin upgrade. Maybe it was deleted in the process? 

Bye and thanks.

```
[2021-08-23 11:44:05.417] [info] Backup started
[2021-08-23 11:44:05.421] [verbose] loadSettings
[2021-08-23 11:44:05.423] [verbose] loadBackupPath
[2021-08-23 11:44:05.432] [verbose] Plugin settings:
[2021-08-23 11:44:05.433] [verbose] path: y:\Research and Development\Working Area\XXXXXXXX\joplin-bkp\
[2021-08-23 11:44:05.434] [verbose] singleJex: true
[2021-08-23 11:44:05.437] [verbose] backupRetention: 3
[2021-08-23 11:44:05.440] [verbose] backupInterval: 24
[2021-08-23 11:44:05.442] [verbose] onlyOnChange: true
[2021-08-23 11:44:05.443] [verbose] usePassword: true
[2021-08-23 11:44:05.445] [verbose] lastBackup: 1626252674049
[2021-08-23 11:44:05.447] [verbose] fileLogLevel: verbose
[2021-08-23 11:44:05.448] [verbose] zipArchive: no
[2021-08-23 11:44:05.450] [verbose] compressionLevel: 0
[2021-08-23 11:44:05.451] [verbose] exportPath: 
[2021-08-23 11:44:05.453] [verbose] backupSetName: {YYYY}-{MM}-{DD}-{HHmm}
[2021-08-23 11:44:05.454] [verbose] backupInfo: [{"name":"2021-07-14-1051","date":1626252674049},{"name":"2021-07-13-1049","date":1626166142194},{"name":"2021-07-12-1046","date":1626079561451}]
[2021-08-23 11:44:05.456] [verbose] activeBackupPath: C:\Users\XXXXXXXX\Joplin\JoplinProfile\tmp\joplin_active_backup_job
[2021-08-23 11:44:05.457] [verbose] backupBasePath: y:\Research and Development\Working Area\XXXXXXXX\joplin-bkp\
[2021-08-23 11:44:05.457] [verbose] logFile: C:\Users\XXXXXXXX\Joplin\JoplinProfile\cache\io.github.jackgruber.backup\activeBackup.log
[2021-08-23 11:44:05.458] [verbose] showDoneMsg: true
[2021-08-23 11:44:05.459] [verbose] installationDir: C:\Users\XXXXXXXX\Joplin\JoplinProfile\cache\io.github.jackgruber.backup
[2021-08-23 11:44:05.464] [info] Enable password protection: true
[2021-08-23 11:44:05.465] [verbose] Backup path: y:\Research and Development\Working Area\XXXXXXXX\joplin-bkp\
[2021-08-23 11:44:05.465] [verbose] Active backup path (export path): C:\Users\XXXXXXXX\Joplin\JoplinProfile\tmp\joplin_active_backup_job
[2021-08-23 11:44:05.466] [verbose] Create folder C:\Users\XXXXXXXX\Joplin\JoplinProfile\tmp\joplin_active_backup_job
[2021-08-23 11:44:05.468] [info] Backup Profile Data
[2021-08-23 11:44:05.468] [verbose] Create folder C:\Users\XXXXXXXX\Joplin\JoplinProfile\tmp\joplin_active_backup_job\profile
[2021-08-23 11:44:05.471] [verbose] Copy C:\Users\XXXXXXXX\Joplin\JoplinProfile\settings.json
[2021-08-23 11:44:05.476] [verbose] Copy C:\Users\XXXXXXXX\Joplin\JoplinProfile\keymap-desktop.json
[2021-08-23 11:44:05.478] [verbose] Copy C:\Users\XXXXXXXX\Joplin\JoplinProfile\userchrome.css
[2021-08-23 11:44:05.480] [verbose] Copy C:\Users\XXXXXXXX\Joplin\JoplinProfile\userstyle.css
[2021-08-23 11:44:05.483] [info] No templateDir, Joplin >= v2.2.5
[2021-08-23 11:44:05.484] [info] Select notebooks for export
[2021-08-23 11:44:05.489] [verbose] Add 'LAVORO' (651d3340ca254cbb9f1574436e6b4761)

[---8<------8<------8<------8<--- SNIP! ---8<------8<------8<------8<---]

[2021-08-23 11:44:05.584] [verbose] Add 'Imported Templates - 23/08/2021' (adf9634c660a49efa969f2b6160045cc)
[2021-08-23 11:44:05.585] [info] Create single file JEX backup
[2021-08-23 11:44:16.258] [info] Create zip archive
[2021-08-23 11:44:16.260] [verbose] Add C:\Users\XXXXXXXX\Joplin\JoplinProfile\tmp\joplin_active_backup_job\all_notebooks.jex to zip C:\Users\XXXXXXXX\Joplin\JoplinProfile\tmp\joplin_active_backup_job\all_notebooks.jex.7z
```

###  UPDATE!
The backup seemed impossible to restart and the password kept disappearing. 
It all seemed frozen.
As soon as I restarted Joplin (several times) the automatic backup started, but it was impossible to finish and the password kept disappearing.

To try to restart from a "clean" situation, I removed the check from "Password protected backups", I stopped Joplin, I deleted the backup logfile and the tmp directory, then I restarted joplin and the first automatic backup started and went successful.
So I reset the "Password protected backups", entered the password, launched a manual backup and everything was fine.
The only strange thing, no done message was showed.

**Comment by JackGruber** [comment]
Please create a Joplin debug log. Which includes the Joplin start and the creation of the backup.
https://joplinapp.org/debugging/#desktop-application

Then please attach the log and console output as a file.

**Comment by JackGruber** [comment]
Could you please check the [pre-release v1.0.4](https://github.com/JackGruber/joplin-plugin-backup/releases/tag/v1.0.4). You have to install it by hand.
This update should prevent a hang.

Eventuel this goes in the same direction as https://github.com/JackGruber/joplin-plugin-backup/issues/22

**Comment by antoniopaolini** [comment]
Done! It works!
I did a manual backup and it went fine.
Nice job, thanks!

I think the only thing missing is a message that the backup was successful (as in the error messages, where maybe you could put an icon like this:  ![image](https://user-images.githubusercontent.com/60228211/131314453-50a4cc9f-1a09-43b5-8019-654dfd366ad1.png) )

**Comment by JackGruber** [comment]
There is no status area where I can display something and a meassage box during an automatic backup is more than annoying.
For a manual backup there should be a message after the backup.

But be careful, your backups should now be unvershclüsselt, because this fix nothing to the problem. Since this is probably not a problem of the plugin. The fields with the passwords contain no text (After a restart of joplin?) and therefore no password is set.

**Comment by antoniopaolini** [comment]
>  The fields with the passwords contain no text (After a restart of joplin?) and therefore no password is set.
You are right!
My backup are password-protected, but today I restarted Joplin and i got a password error:
![image](https://user-images.githubusercontent.com/60228211/131631907-2624928f-2e7a-4646-bd34-6bb61b1ebf0d.png)

Then, where's the problem?

**Comment by JackGruber** [comment]
Eventuel this goes in the same direction as #22 and if this is so, there is a bug in Joplin and not in the plugin. But since it does not occur in my case, I can not recreate or analyze this.

**Comment by antoniopaolini** [comment]
Hello Jack, 
The problem, as you know, persists.
Every time I restart Joplin the password are gone. I need to go in Option->backup and rewrite the password (despite I see asterisks in the textbox).

This morning it occurred to me to try looking in the database.
Right after relaunching Joplin this is what I see:
![image](https://user-images.githubusercontent.com/60228211/141460758-d43d7564-cf20-4bb8-90fa-2e84be30a638.png)

Really "strange", don't you think?

Then I reinserted, as usual, the password, I updated the query and then I see the correct password in both the fields:
![image](https://user-images.githubusercontent.com/60228211/141462273-dc0c6915-218b-4b87-ac9a-7cf7973f7726.png)

Another piece of information that I don't remember if I already gave it to you: I use the portable version of Joplin

Does this information help?
Bye.

**Comment by JackGruber** [comment]
>Really "strange", don't you think?
No this passwords are the default passwords, when there is a password check error. 

I think there is a bug in Joplin with password fields an installation which do not support KeyChain (Like portable App).

**Comment by r4pr0n** [comment]
> I think there is a bug in Joplin with password fields an installation which do not support KeyChain (Like portable App).
Well personally I don't run this as a portable app but it still occurs. I'm not sure what kind of keychain you mean?
I have a gnome secret service daemon, ssh-agent and gpg-agent running.

**Comment by JackGruber** [comment]
> I'm not sure what kind of keychain you mean?

Keychain is a secure credential manager from the OS and Joplin stores the passwords via this if this is supported.

You can check in `About Joplin` if you have keychain support or not
![grafik](https://user-images.githubusercontent.com/24863925/141645635-b0a47095-6fc5-4e2f-8c68-985f9f1bb702.png)

**Comment by r4pr0n** [comment]
Oh yes, you are right, that is also disabled for me. Then this probably is the issue.

**Comment by JackGruber** [comment]
I have created an issue for the problem when no keychain support is available https://github.com/laurent22/joplin/issues/5720

**Comment by antoniopaolini** [comment]
![image](https://user-images.githubusercontent.com/60228211/141677672-078b76e6-dcf2-4dfb-bd13-01bcf8eb330c.png)
Yes, it' also my case!

Thanks for creating the issue

**Comment by JackGruber** [comment]
The problem with no KeyChain support and lost password is fixed in Joplin [2.6.2](https://github.com/laurent22/joplin/releases/tag/v2.6.2)

**Comment by antoniopaolini** [comment]
A small comment, just to confirm that with 2.6.2 it works! 🎉
Thanks! 🍻
